eth-contracts/contracts/DelegateManager.sol

```diff
@@ -430,6 +404,49 @@ contract DelegateManager is RegistryContract {
         );
     }
 
+    function _validateClaimRewards(address _sp)
```

Comment:
just took a stab at breaking up a lot of the pre processing. i think we can be more aggressive if we re-order stuff, like put the ` if (totalBalanceInStaking == totalBalanceOutsideStaking) {` check after defining totalRewards and totalActiveFunds